src/riscv/lib/src/pvm/tezos.rs

```diff
@@ -43,10 +45,53 @@ use crate::machine_state::registers::a1;
 use crate::machine_state::registers::a2;
 use crate::machine_state::registers::a3;
 use crate::machine_state::registers::a6;
+use crate::pvm::linux::ECALL_WIDTH;
 use crate::state_backend::Cell;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::ManagerWrite;
 
+/// The result of handling a host function.
+/// This is similar to `linux::parameters SystemCallResultExecution` but there
+/// are slight differences between what we allow tezos and linux calls to do.
+struct TezosCallResult {
+    /// The value written to output register `a0`.
+    /// Functions such as `tezos_inbox_next` don't write register hence the `Option`.
+    result: Option<Result<XValue, SbiError>>,
+
+    /// Usually, handling a host function involves an increment to the pc.
+    pc_update: ProgramCounterUpdate<Address>,
+}
+
+/// Having this trait from a Result type allows most handlers to use the `?` syntax
+impl<T: Into<u64>> From<Result<T, SbiError>> for TezosCallResult {
+    fn from(value: Result<T, SbiError>) -> Self {
+        // The default action is to continue execution after the host function call.
+        // In cases where the execution should halt, this should be specified.
+        Self {
+            result: Some(value.map(Into::into)),
+            pc_update: ProgramCounterUpdate::Next(ECALL_WIDTH),
+        }
+    }
+}
+
+impl From<SbiError> for TezosCallResult {
+    fn from(value: SbiError) -> Self {
+        // The default action is to continue execution after the host function call.
+        // In cases where the execution should halt, this should be specified.
+        Self {
+            result: Some(Err(value)),
+            pc_update: ProgramCounterUpdate::Next(ECALL_WIDTH),
+        }
+    }
+}
+
+impl TezosCallResult {
+    const NO_RESULT: Self = TezosCallResult {
```

Comment:
Technically Tezos host functions always have results. It's just that those are populated later. This seems important information to convey.

```suggestion
    const SUSPENDED: Self = TezosCallResult {
```

Reply:
Done [here](https://github.com/tezos/riscv-pvm/compare/bbd664db30806cc1bf6bf9bfc34ce9013257d30b..37368f2b5f08780e9da54d69ae157e07661d6ec5)